django-verdant/rca/api/filters.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+from rest_framework import filters
+
+from django.core.exceptions import FieldDoesNotExist
+from django.db.models.fields.related import ForeignObjectRel, OneToOneRel
+from rca.models import EventItemDatesTimes, EventItem
+
+class RelatedProgrammesFilter(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, 'related_programmes'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(related_programmes__programme__slug__in=rp)
+
+        if hasattr(queryset.model, 'related_programme'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(related_programme__slug__in=rp)
+
+        if hasattr(queryset.model, 'roles'):
+            rp = request.GET.getlist('rp', [])
+            if rp:
+                queryset = queryset.filter(roles__programme__slug__in=rp)
+
+        return queryset
+
+
+class NextEventOrder(filters.OrderingFilter):
+    def filter_queryset(self, request, queryset, view):
+        if hasattr(queryset.model, 'dates_times'):
+            date_from = request.query_params.get('event_date_from', None)
+            if date_from:
+                now = datetime.now()
+                now = datetime.date(now)
```

Comment:
`datetime.now().date()`? If you comparing dates, do you need to care about the timezone? You may want to use `datetime.utcnow().date()`. I guess it's not critical.